internal/service/advancedcluster/resource_update_logic.go

```diff
@@ -0,0 +1,61 @@
+package advancedcluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"go.mongodb.org/atlas-sdk/v20240530002/admin"
+)
+
+func noIDsPopulatedInReplicationSpecs(replicationSpecs *[]admin.ReplicationSpec20250101) bool {
```

Comment:
Can we add comments about "why" these methods are used? maybe mention which scenarios need these?

Would be helpful to capture API behavior here for future reference

Reply:
left a comment in the single usage of this function providing context, this function by itself is trivial so thought a comment was more valuable for the place where it is called.

Reply:
makes sense, thanks!